pkg/k8s/watchers/watcher.go

```diff
@@ -731,6 +726,15 @@ func (k *K8sWatcher) delK8sSVCs(svc k8s.ServiceID, svcInfo *k8s.Service, se *k8s
 				frontends = append(frontends, loadbalancer.NewL3n4Addr(svcPort.Protocol, addrCluster, svcPort.Port, loadbalancer.ScopeInternal))
 			}
 		}
+
+		if svcInfo.IsHeadless {
+			for addr := range se.Backends {
+				frontends = append(frontends, loadbalancer.NewL3n4Addr(svcPort.Protocol, addr, svcPort.Port, loadbalancer.ScopeExternal))
+				if svcInfo.ExtTrafficPolicy == loadbalancer.SVCTrafficPolicyLocal || svcInfo.IntTrafficPolicy == loadbalancer.SVCTrafficPolicyLocal {
+					frontends = append(frontends, loadbalancer.NewL3n4Addr(svcPort.Protocol, addr, svcPort.Port, loadbalancer.ScopeInternal))
+				}
```

Comment:
The additional entry with internal scope seems to be created only when the service type is either LB or NodePort \[1\]. It should be mostly harmless though, maybe causing only the emission of a warning log.

\[1\]: https://github.com/cilium/cilium/blob/f66f4b159d24e1b5c8e4d92a69932ef003cff87d/pkg/k8s/watchers/watcher.go#L761-L763